src/core/defs.jl

```diff
@@ -754,55 +754,39 @@ function import_params!(comp::AbstractComponentDef)
-              comp_name::Symbol=comp_def.comp_id.comp_name; rename=nothing,
-              first=nothing, last=nothing, before=nothing, after=nothing)
+    add_comp!(
+        obj::AbstractCompositeComponentDef, 
+        comp_def::AbstractComponentDef,
+        comp_name::Symbol=comp_def.comp_id.comp_name; 
+        before::NothingSymbol=nothing, 
+        after::NothingSymbol=nothing,
+        rename::NothingPairList=nothing
+    )
 
 Add the component `comp_def` to the composite component indicated by `obj`. The component is 
 added at the end of the list unless one of the keywords `before` or `after` is specified.
 Note that a copy of `comp_id` is made in the composite and assigned the give name. The optional
 argument `rename` can be a list of pairs indicating `original_name => imported_name`.
-
-Note: `first` and `last` keywords are currently disabled.
 """
-function add_comp!(obj::AbstractCompositeComponentDef, comp_def::AbstractComponentDef,
-                   comp_name::Symbol=comp_def.comp_id.comp_name;
-                   first::NothingInt=nothing, last::NothingInt=nothing,
-                   before::NothingSymbol=nothing, after::NothingSymbol=nothing,
-                   rename::NothingPairList=nothing)
-
-    if first !== nothing || last !== nothing
-        @warn "add_comp!: Keyword arguments 'first' and 'last' are currently disabled."
-        first = last = nothing
-    end
-
-    # When adding composites to another composite, we disallow setting first and last periods.
-    if is_composite(comp_def) && (first !== nothing || last !== nothing)
-        error("Cannot set first or last period when adding a composite component: $(comp_def.comp_id)")
-    end
+function add_comp!(
+    obj::AbstractCompositeComponentDef, 
+    comp_def::AbstractComponentDef,
+    comp_name::Symbol=comp_def.comp_id.comp_name;
+    before::NothingSymbol=nothing, 
+    after::NothingSymbol=nothing,
+    rename::NothingPairList=nothing
+)
 
     # Check if component being added already exists
     has_comp(obj, comp_name) && error("Cannot add two components of the same name ($comp_name)")
 
+    if before !== nothing && after !== nothing
+        error("Cannot specify both 'before' and 'after' parameters")
+    end
+
     # check time constraints if the time dimension has been set
     if has_dim(obj, :time)
         # error("Cannot add component to composite without first setting time dimension.")
-
-        # check that first and last are within the model's time index range
-        time_index = time_labels(obj)
-
-        if first !== nothing && first < time_index[1]
-            error("Cannot add component $comp_name with first time before first of model's time index range.")
-        end
-
-        if last !== nothing && last > time_index[end]
-            error("Cannot add component $comp_name with last time after end of model's time index range.")
-        end
-
-        if before !== nothing && after !== nothing
```

Comment:
I think the `before` and `after` stuff should remain, right?

Reply:
I moved it outside of the `if` statement, since I think it should always be checked, not just when there is a time dimension? (although I think all components have a time dimension, so I think it was always being checked)

Reply:
@rjplevin is this okay?

Reply:
Yes, I think so. (Honestly, I haven't looked very closely, but I trust you've thought it through!)